File: case02/java/Mercado.java
public class Mercado
{
    private Prateleira prateleira1, prateleira2;
    
    public Mercado(Prateleira prateleira1, Prateleira prateleira2){
        this.prateleira1 = prateleira1;
        this.prateleira2 = prateleira2;
    }
    public Prateleira getPrateleira1(){
        return prateleira1;
    }
    public Prateleira getPrateleira2(){
        return prateleira2;
    }
    public double mediaValorProdutos(){
        return (prateleira1.getProduto1().getPreco()+prateleira1.getProduto2().getPreco()+prateleira1.getProduto3().getPreco())/3;
}
}

File: case02/python/Mercado.py
from Prateleira import Prateleira

class Mercado:
    def __init__(self, prat1: Prateleira, prat2: Prateleira):
        self.prat1 = prat1
        self.prat2 = prat2

    def get_prat1(self):
        return self.prat1
    
    def get_prat2(self):
        return self.prat2
    
    def __str__(self):
        return f"1 - {self.prat1.__str__()}\n2 - {self.prat2.__str__()}"
    
    def mediaValorProdutos(self):
        return (self.prat1.get_prod1().get_preco()+self.prat1.get_prod2().get_preco()+self.prat1.get_prod3().get_preco()+self.prat2.get_prod1().get_preco()+self.prat2.get_prod2().get_preco()+self.prat2.get_prod3().get_preco())/6